backends.js

```diff
@@ -0,0 +1,47 @@
+const Lynx = require('lynx');
+const cloudwatchMetrics = require('cloudwatch-metrics');
+
+module.exports = {
+  custom: function createCustomBackend(config) {
+    if (!config.send) {
+      throw new Error('Custom backend must expose send(type, name, data) {} method');
+    }
+    return config;
+  },
+  statsd: function createStatsDBackend(config) {
+    if (!config.host || !config.port) {
+      throw new Error('StatsD backend must missing config params');
+    }
+    const client = new Lynx(config.host, config.port);
+    return {
+      send: (type, name, data) => {
+        if (!client[type]) throw new Error(`StatsD backend does not support method '${type}'`);
+        client[type](name, data);
+      },
+    };
+  },
+  cloudwatch: function createCloudwatchBackend(config) {
+    if (!config.region || !config.groupName || !config.accessKeyId || !config.secretAccessKey) {
```

Comment:
Just a suggestion in case you want to reduce the line width a bit:
```js
if (['region', 'groupName', 'accessKeyId', 'secretAccessKey'].every(key => config[key]) {
  ...
}
```